File: src/main/java/com/koddy/server/auth/infrastructure/oauth/kakao/response/KakaoTokenResponse.java
package com.koddy.server.auth.infrastructure.oauth.kakao.response;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import com.koddy.server.auth.domain.model.oauth.OAuthTokenResponse;

@JsonNaming(value = PropertyNamingStrategies.SnakeCaseStrategy.class)
public record KakaoTokenResponse(
        String tokenType,
        String accessToken,
        String refreshToken,
        long refreshTokenExpiresIn,
        long expiresIn
) implements OAuthTokenResponse {
}

File: src/main/java/com/koddy/server/auth/infrastructure/social/kakao/response/KakaoTokenResponse.kt
package com.koddy.server.auth.infrastructure.social.kakao.response

import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming
import com.koddy.server.auth.domain.model.oauth.OAuthTokenResponse

@JsonNaming(value = PropertyNamingStrategies.SnakeCaseStrategy::class)
data class KakaoTokenResponse(
    val tokenType: String,
    val accessToken: String,
    val refreshToken: String?,
    val refreshTokenExpiresIn: Long?,
    val expiresIn: Long,
) : OAuthTokenResponse {
    override fun accessToken(): String = accessToken
}
